compiler/resolution/functionResolution.cpp

```diff
@@ -7406,7 +7406,17 @@ static void handleTaskIntentArgs(CallInfo& info, FnSymbol* taskFn) {
 
       // Need to copy varActual->type even for type variables.
       // BTW some formals' types may have been set in createTaskFunctions().
-      formal->type = varActual->type;
+      // If we're performing a copy, the type might change (e.g., array view
+      // becomes array). In that case, make the formal type be the post-copy type.
+      auto determineType = [formal](Type* t) {
+        Type* copyType;
+        if (inOrOutFormalNeedingCopyType(formal) &&
+            (copyType = getCopyTypeDuringResolution(t->getValType()))) {
+          return copyType;
+        }
```

Comment:
the control flow on this could be written cleaner. This is concise but less clear. Assigning and using that value in the conditional is too subtle for clear code imo

Consider
```
if (inOrOutFormalNeedingCopyType(formal)) {
  if (Type*  copyType = getCopyTypeDuringResolution(t->getValType()))
    return copyType;
}
return t;
```